pkg/templates/nstemplatetiers/nstemplatetier_generator.go

```diff
@@ -52,3 +40,45 @@ func CreateOrUpdateResources(ctx context.Context, s *runtime.Scheme, client runt
 		return applyCl.ApplyObject(ctx, toEnsure, commonclient.ForceUpdate(true))
 	}, namespace, metadata, files)
 }
+
+func LoadFiles(nsTemplateTiers embed.FS, root string) (metadata map[string]string, files map[string][]byte, err error) {
```

Comment:
Maybe add some comment here about the assumptions on the directory structure in the supplied FS?

Also, if the tests were in the same package as this file, this function could be private which again would improve the code cohesion and lower the possibility of misuse.

Reply:
added the comments.
i did try to make the func private but the test package didnt pick it up. and hence kept it as is

Reply:
The reason why the test didn't see the function when you made it private is because of the fact that, for some reason that I don't fully understand, the test lives in another package, `nstemplatetier_test`, than the code itself - `nstemplatetier`. If the test file declared the same package as the the code, you'd be able to test the private functions, too.

Reply:
yes but  the reason, we cannot do that is, the import cycles aren't allowed in test, and we do need them in our test so we cannot have the same package